src/libraries/System.Reflection.Emit/src/System/Reflection/Emit/SignatureHelper.cs

```diff
@@ -294,6 +294,7 @@ private static void WriteSimpleSignature(SignatureTypeEncoder signature, Type ty
         }
     }
 
+    // The order of the enum values should match with the ModuleBuilderImpl.s_coreTypes array elements order.
```

Comment:
Late to the party, but would moving the `s_coreTypes` array here from _ModuleBuilderImpl.cs_ and exposing internally be safer to ensure they are kept in sync? Since this SignatureHelper.cs is part of the same assembly and is a pure static class, there should be no actual difference in allocation/initialization behavior.

Reply:
No, as you mentioned `SignatureHelper` is static, core types should be populated for each `ModuleBuilder` instance with the user assembly provided, and it is not only used for signature, for example it is used for loading parent type of reference when creating `TypeBuilder`

Reply:
Perfect, just wanted to see if putting them in the same place in source would be clearer. Thanks!